Signal/src/view controllers/OversizeTextMessageViewController.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import WebRTC
+import PromiseKit
+
+class OversizeTextMessageViewController: UIViewController {
+
+    let TAG = "[OversizeTextMessageViewController]"
+
+    let message: TSMessage
+
+    // MARK: Initializers
+
+    @available(*, unavailable, message:"use message: constructor instead.")
+    required init?(coder aDecoder: NSCoder) {
+        message = TSMessage()
+        super.init(coder: aDecoder)
+    }
+
+    required init(message: TSMessage) {
+        self.message = message
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    // MARK: Attachment
+
+    private func attachmentStream() -> TSAttachmentStream? {
+        guard message.hasAttachments() else {
+            Logger.error("\(TAG) message has no attachments.")
+            assert(false)
+            return nil
+        }
+        let attachmentIDRaw = message.attachmentIds[0]
+        guard attachmentIDRaw is String else {
+            Logger.error("\(TAG) message attachment id is not a string.")
+            assert(false)
+            return nil
+        }
+        let attachmentID = attachmentIDRaw as! String
+        guard let attachment = TSAttachment.fetch(withUniqueID:attachmentID) else {
+            Logger.error("\(TAG) could not load attachment.")
+            assert(false)
+            return nil
+        }
+        guard attachment.contentType == OWSMimeTypeOversizeTextMessage else {
+            Logger.error("\(TAG) attachment has unexpected content type.")
+            assert(false)
+            return nil
+        }
+        guard attachment is TSAttachmentStream else {
```

Comment:
nit: could combine a couple of these guards to be more concise and avoid the `!`
```
guard let stream = TSAttachment.fetch(withUniqueID:attachmentID) as? TSAttachmentStream else {
...
}
```